### PR TITLE
DOC: Plot discrete colormap with proportional colorbar

### LINKED ISSUE #1015: 'levels' tick labels for discrete colormaps
Hi there,

'levels' tick labels for discrete colormaps works fine for 10 intervals and tend to shorten by itself beyond 10. This behavior may exist to make colorbar exceeding 10 intervals look tidy.

However, for **discrete** colormaps it will be good to label the ticks for all intervals as they can be any number and not uniform.
xray.**version** = '0.8.2'

Please see the screenshots attached for a better understanding.

Thanks,
Vijay Paul
image

image
image

### PR DESCRIPTION
**Pull request by paulvija**
Using 'cbar_kwargs' it is possible to extend the capabilities of colorbar ticks.

 - [x] Closes #1015 (remove if there is no corresponding issue, which should only be the case for minor changes)

### CODE REVIEW

**Comment by shoyer** [comment]
Apparently we did not successfully disable stickler in doc directories.

But please do fix the line-length issue. I don't think using an absolute import would improve clarity here.

File: doc/gallery/plot_proportional_colorbar.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+"""
+============================================
+Proportional colorbar for discrete colormaps
```
Comment (fmaussion) [inline doc/gallery/plot_proportional_colorbar.py]: Maybe: ``Control the plot's colorbar``

File: doc/gallery/plot_proportional_colorbar.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+============================================
+
+Use ``cbar_kwargs`` keyword to specify the number of ticks.
+``spacing`` can be used to draw the ticks proportional too.
```
Comment (fmaussion) [inline doc/gallery/plot_proportional_colorbar.py]: ... ``to draw proportional ticks.``

File: doc/gallery/plot_proportional_colorbar.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+"""
+from __future__ import absolute_import
```
Comment (fmaussion) [inline doc/gallery/plot_proportional_colorbar.py]: Sorry, please ignore our bot on this one: remove this line

File: doc/gallery/plot_proportional_colorbar.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import xarray as xr
+import matplotlib.pyplot as plt
+
+# Load the data
+air_temp = xr.tutorial.load_dataset('air_temperature')
+air = air_temp.air
+air2d = air.isel(time=500)
```
Comment (fmaussion) [inline doc/gallery/plot_proportional_colorbar.py]: merge in one line: 

```
air2d = air_temp.air.isel(time=500)
```

File: doc/gallery/plot_proportional_colorbar.py (hunk 3 above)
Comment (shoyer) [reply doc/gallery/plot_proportional_colorbar.py]: Yes, the bot has been terminated -- if you merge in master, it should no longer bother you!

File: doc/gallery/plot_proportional_colorbar.py (hunk 1 above)
Comment (paulvija) [reply doc/gallery/plot_proportional_colorbar.py]: Makes more sense.

**Comment by paulvija** [comment]
@shoyer Looks like stickler bot is still active.

**Comment by fmaussion** [comment]
Thanks @paulvija !

**Comment by paulvija** [comment]
@fmaussion Would the plot image display automatically here or something I have to do at my end?
http://xarray.pydata.org/en/latest/auto_gallery/control_plot_colorbar.html

**Comment by shoyer** [comment]
Hmm, I'm not sure. Let's see if rebuilding the docs fixes it...

If that doesn't work, it might be a good idea to do some tests with a local docs build.

**Pull request merged**